File: solutions/python/0009.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        reversed = 0
        original = x
        while x > 0:
            reversed = reversed * 10 + x % 10
            x = x // 10
        return reversed == original

File: solutions/rust/0009.rs
impl Solution {
    fn is_palindrome(mut x: i32) -> bool {
        if x < 0 {
            return false;
        }
        let mut reversed = 0;
        let mut original = x;
        while x > 0 {
            reversed = reversed * 10 + x % 10;
            x = x / 10;
        }
        return reversed == original;
    }
}